src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,73 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class EventFrequencyPercentForm(EventFrequencyForm):
+    intervals = percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class EventFrequencyPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = percent_intervals
+        self.form_cls = EventFrequencyPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        cache_key = f"r.c.spc:{environment_id}-{start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}-{end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
+            " ", ""
+        )
+        session_count = cache.get(cache_key)
+        if session_count is None:
+            filters = {"project_id": [event.project_id]}
+            if environment_id:
+                filters["environment"] = [environment_id]
+            result_totals = raw_query(
+                selected_columns=["sessions"],
+                rollup=60,
```

Comment:
I'm a little confused by this rollup as well: if there is an interval of 1 hour, then you are saying to group sessions by `toStartOfMinute(bucketed_started)`, which means you'll get up to 60 results back. You're then only selecting the first value below. If you only need one minute worth of data, why query for all 60?

Reply:
Hmm, so I made the rollup 60 so that we'd hit the raw table. When I did that, I definitely needed to start summing the results in every bucket (at the time of your comment, it would have only been getting correct counts for 1m intervals). Just fixed that. Previously there was no bucketing and only one result of the session count, so I didn't sum them up.

It sounds like you would prefer I use a rollup of 3600. If I make the rollup 3600, snuba groups using `toStartOfHour` which works, but it also starts hitting the hourly table.

So if I need one minute of data from the raw table, right now the only way I know of to do this is to make the rollup 60 and use start/end to limit the query. If start/end are 1 minute apart, I only seem to get one bucket back. If they are 10 minutes apart, I seem to get max ten back (if there were sessions every minute). This seems basically fine to me.

So now I sum up the results, and I modified the test to better spread the sessions throughout the time period so that without summing it will actually fail. What do you think?

Reply:
Now that you're using all the results it's fine. I only mentioned a limit because you were only using 1 row of data even though you were getting back more than that. The 60 second rollup is fine.